torch/_inductor/decomposition.py

```diff
@@ -444,16 +444,16 @@ def view_copy_dtype(self, dtype):
     decomp.register_decomposition, registry=extra_random_decomps
 )
 
+if utils.has_triton():
 
-@register_extra_random_decomp([aten.bernoulli_])
-def bernoulli_(self, p=0.5):
-    return self.copy_(torch.rand_like(self, dtype=torch.float32) < p)
+    @register_extra_random_decomp([aten.bernoulli_])
+    def bernoulli_(self, p=0.5):
+        return self.copy_(torch.rand_like(self, dtype=torch.float32) < p)
 
-
-@register_extra_random_decomp([aten.bernoulli.p])
-def bernoulli_p(self, p=0.5, *, generator=None):
-    assert generator is None
-    return torch.rand_like(self, dtype=torch.float32) < p
+    @register_extra_random_decomp([aten.bernoulli.p])
+    def bernoulli_p(self, p=0.5, *, generator=None):
+        assert generator is None
+        return torch.rand_like(self, dtype=torch.float32) < p
```

Comment:
It is possible both cpu and gpu are available while we only want to do fallback on cpu. Suggest to check the device type of `self` and raise `NotImplemented` to go directly to lowering if the device is cpu.

Reply:
Thanks for your suggestion. I have updated all checks.